designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala

```diff
@@ -210,15 +216,25 @@ class ScenarioActivityApiHttpService(
       )
 
   private def fetchActivities(
-      scenarioId: ProcessId
-  ): EitherT[Future, ScenarioActivityError, List[Dtos.ScenarioActivity]] =
-    EitherT
-      .right(
-        dbioActionRunner.run(
-          scenarioActivityRepository.findActivities(scenarioId)
-        )
-      )
-      .map(_.map(toDto).toList)
+      processIdWithName: ProcessIdWithName
+  )(implicit loggedUser: LoggedUser): EitherT[Future, ScenarioActivityError, List[Dtos.ScenarioActivity]] = {
+    val combined = for {
+      generalActivities <- dbioActionRunner.run(scenarioActivityRepository.findActivities(processIdWithName.id))
+      deploymentManager <- deploymentManagerDispatcher.deploymentManager(processIdWithName)
+      deploymentManagerSpecificActivities <- deploymentManager match {
+        case Some(manager) =>
+          manager.scenarioActivityHandling match {
+            case AllScenarioActivitiesStoredByNussknacker =>
+              Future.successful(List.empty)
+            case handling: ManagerSpecificScenarioActivitiesStoredByManager =>
+              handling.managerSpecificScenarioActivities(processIdWithName)
+          }
+        case None =>
+          Future.successful(List.empty)
+      }
+    } yield generalActivities ++ deploymentManagerSpecificActivities
+    EitherT.right(combined).map(_.map(toDto).toList.sortBy(_.date))
```

Comment:
maybe we should move the DTO creation and sorting to the yield?
at the moment it's rather hidden in the monad transformer boilerplate